core/src/impl/Kokkos_ViewMapping.hpp

```diff
@@ -721,6 +721,32 @@ struct SubviewExtents {
   }
 };
 
+template <unsigned TrivialScalarSize>
+struct Padding {
+  // If memory alignment is a multiple of the trivial scalar size then attempt
+  // to align.
+  KOKKOS_INLINE_FUNCTION
+  static constexpr size_t stride(size_t const N) {
+    if constexpr (TrivialScalarSize == 0) {
+      return N;
+    } else {
+      constexpr int div   = Kokkos::Impl::MEMORY_ALIGNMENT / TrivialScalarSize;
+      constexpr int mod   = Kokkos::Impl::MEMORY_ALIGNMENT / TrivialScalarSize;
```

Comment:
```suggestion
      constexpr int mod   = Kokkos::Impl::MEMORY_ALIGNMENT % TrivialScalarSize;
```

Reply:
Also, if the tests didn't/don't catch that, then it exposes a gap in the tests

Reply:
Yes, I'll add a check that that Padding has effect.